File: backend/app/schemas/market.py
from datetime import datetime
from decimal import Decimal
from uuid import UUID

from pydantic import BaseModel


class KlineRequest(BaseModel):
    symbol: str
    market: str
    timeframe: str = "1d"
    start: str | None = None
    end: str | None = None
    limit: int = 500


class KlineData(BaseModel):
    timestamp: datetime
    open: Decimal
    high: Decimal
    low: Decimal
    close: Decimal
    volume: Decimal


class SymbolInfo(BaseModel):
    symbol: str
    name: str
    market: str

File: frontend/src/types/market.ts
export interface SymbolInfo {
  symbol: string;
  name: string;
  market: string;
}

export interface KlineData {
  timestamp: string;
  open: string;
  high: string;
  low: string;
  close: string;
  volume: string;
}

export interface TickPrice {
  symbol: string;
  price: string;
  timestamp: string;
}

export interface WatchlistItem {
  id: string;
  symbol: string;
  market: string;
  sort_order: number;
}
